### PR TITLE
Address Changes in Evnironment API

### PR DESCRIPTION
**Pull request by linawolf**
And further changes regarding directory `typo3conf`

references https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/204
resolves https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/issues/2287

### CODE REVIEW

File: Documentation/ApiOverview/Environment/Index.rst [hunk 1]
```diff
@@ -63,29 +63,47 @@ getVarPath()
 The environment provides the path to the :file:`var` folder. This folder contains
 data like logs, sessions, locks, and cache files.
 
-For projects with Composer setup, the value is :php:`getProjectPath() . '/var'`,
-so it is outside of the web document root - not within :php:`getPublicPath()`.
+For Composer-based installations, the it returns :ref:`directory-var`, in legacy
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:66]:
```suggestion
For Composer-based installations, it returns :ref:`directory-var`, in legacy
```

File: Documentation/Security/GuidelinesIntegrators/InstallTool.rst [hunk 2]
```diff
@@ -30,6 +30,12 @@ need write access to this directory on a server level (for example via SSH, SFTP
 or you can create this file as a backend user with administrator
 privileges.
 
+..  attention::
+    Even if you run TYPO3 v12 in the Composer-mode at the time of writing
```
Comment (brotkrueml) [inline Documentation/Security/GuidelinesIntegrators/InstallTool.rst:34]:
```suggestion
    Even if you run TYPO3 v12 in Composer mode at the time of writing
```

Had to read it multiple times, it sounds strange without some punctuation, maybe better:

Even if you run TYPO3 v12 in the Composer-mode, the file has still to be created in folder public/typo3conf/ENABLE_INSTALL_TOOL (as at the time of writing TYPO3 v12.0.0).

But I am not sure if we have to mention it. Do you create the file before accessing the install tool without being logged-in into the backend? I always call install.php, copy the path (as I mostly don't remember the name) and make a "touch" on file system. I think, the most people will do it that way. And I assume this file will be moved into the public/project folder, so this note becomes obsolete later. Maybe we should just add a todo with a link to the forge ticket to pick it up later.

File: Documentation/ApiOverview/Environment/Index.rst [hunk 3]
```diff
@@ -63,29 +63,47 @@ getVarPath()
 The environment provides the path to the :file:`var` folder. This folder contains
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:63]:
```suggestion
The environment provides the path to the :file:`var/` folder. This folder contains
```

File: Documentation/ApiOverview/Environment/Index.rst [hunk 4]
```diff
@@ -63,29 +63,47 @@ getVarPath()
...
-The environment provides the path to :file:`typo3conf`. This folder contains TYPO3
-global configuration files and folders, e.g. :file:`config/system/settings.php`.
+In Composer based installation this method provides the path
+:ref:`directory-config`, in legacy installations
+:ref:`legacy-directory-typo3conf`.
+
+The directory returned by this method contains the folders :file:`system/`
+containing the :ref:`configuration files <configuration-files>`
+:file:`system/settings.php` and :file:`system/additional.php` and the folder
+:file:`sites/` containing the :ref:`site configuration <sitehandling>`.
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:94]:
```suggestion
:file:`sites/` containing the :ref:`site configurations <sitehandling>`.
```

**Review by brotkrueml — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/Environment/Index.rst [hunk 5]
```diff
@@ -63,29 +63,47 @@ getVarPath()
...
+For Composer-based installations, it returns :ref:`directory-var`, in legacy
+installations :ref:`legacy-directory-typo3temp-var`.
 
-Without Composer, the value is :php:`getPublicPath() . '/typo3temp/var'`, so within
-the web document root - a situation that is not optimal from a security point of view.
+..  code-block:: php
+
+    use TYPO3\CMS\Core\Core\Environment;
+
+    // Composer-based installations: '/home/www/my-project/var/`
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/Environment/Index.rst]: done

File: Documentation/ApiOverview/Environment/Index.rst [hunk 6]
```diff
@@ -166,17 +189,17 @@ publicly accessible (for example via web browser) anymore.
 
 A typical example:
 
-- :php:`TYPO3_PATH_APP` is set to :file:`/var/www/my-project`.
-- The web folder is then set to :php:`TYPO3_PATH_ROOT` :file:`/var/www/my-project/public`.
+- :php:`TYPO3_PATH_APP` is set to :file:`/home/www/my-project`.
```
[the remark this answers is not in the document]
Comment (linawolf) [reply Documentation/ApiOverview/Environment/Index.rst]: removed it

File: Documentation/ApiOverview/Environment/Index.rst [hunk 7]
```diff
@@ -6,10 +6,10 @@
 Environment
 ===========
 
-Since version 9.x the TYPO3 Core  includes an environment class.
-This class contains all environment-specific information, e.g. paths within the
+The TYPO3 Core  includes an environment class that s contains all
```
Comment (brotkrueml) [inline Documentation/ApiOverview/Environment/Index.rst:9]:
```suggestion
The TYPO3 Core includes an environment class that contains all
```

**Pull request merged**